README.md

```diff
@@ -416,6 +416,38 @@ server.addHandler( new FileFallbackHandler(_fs, "/jquery/jq1.11.1.js" , "/jquery
 server.addHandler( new FileFallbackHandler(_fs, "/jquery/jqm1.4.5.js" , "/jquery/jqm1.4.5.js" ,   "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js",  "max-age=86400"));
 ```
 
+### Serving static files from SPIFFS
+Serve a specific file or attach a folder in SPIFFS to a root url.
+An optional Cache-Control header can be added to responses.
+An optional Last-Modified header can be added, if the request contains If-Modified-Since header with the same value the server will respond with 304 code instead of serving the file.
+```cpp
+
+// Serve static file example
+server.serveStatic("/url/file.htm", SPIFFS, "/dir/file.htm");
+// Serve static folder example
+server.serveStatic("/root-url", SPIFFS, "/root-dir");
+// Specify cache control
+server.serveStatic("/root-url", SPIFFS, "/root-dir", "max-age=86400");
+// Specify last modified (should be updated each time files are updated)
+server.serveStatic("/root-url", SPIFFS, "/root-dir", NULL, "Wed, 08 Jun 2016 22:00:00 GMT");
+
+```
+
+### HTTP Redirect (302)
+Return HTTP 302 response to redirect the client to a different location.
+An optional IPAddress can specify redirection exclusions, for example, exclude redirection to requests to the AP address.
+```cpp
+// Redirect all requests
+server.redirect("/root/file.ext", "/other-location/other-file.ext");
+// Redirect requests not for the AP address
```

Comment:
What the FileFallbackHandler does is a combination of redirect or serve file, not the best since there is a request to the ESP. The redirect functionality in my pull request does the same but in a more generic way - redirect to CDN if not on AP side, otherwise look for another handler to serve the request.

Using FileFallbackHandler:
server.addHandler( new FileFallbackHandler(SPIFFS, "/jquery/jq1.11.1.js" , "/jquery/jq1.11.1.js" ,   "http://code.jquery.com/jquery-1.11.1.min.js",  "max-age=86400"));

Using my code:
server.redirect("/jquery/jq1.11.1.js", "http://code.jquery.com/jquery-1.11.1.min.js", WiFi.softAPIP());
server.serveStatic("/jquery/jq1.11.1.js", SPIFFS, "/jquery/jq1.11.1.js", "max-age=86400");

However, the serve static can serve folder instead of individual file:
server.redirect("/jquery/jqm1.4.5.css", "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.css", WiFi.softAPIP());
server.redirect("/jquery/jq1.11.1.js", "http://code.jquery.com/jquery-1.11.1.min.js", WiFi.softAPIP());
server.redirect("/jquery/jqm1.4.5.js", "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js", WiFi.softAPIP());
server.serveStatic("/jquery", SPIFFS, "/jquery", "max-age=86400");

Both solutions are based on html page with the following in the header:
    <link rel="stylesheet" href="jquery/jqm1.4.5.css" />
    <script src="jquery/jq1.11.1.js"></script>
    <script src="jquery/jqm1.4.5.js"></script>

What I'm suggesting is to serve 2 different html index pages, one when index.htm is request on AP with the above header, and another when index.htm is requested on STA with header that link directly to the CDN to skip the request to the ESP and the redirection.
Something like:
server.serveStaticOnAP("/", SPIFFS, "/ap");
serverServeStatic("/", SPIFFS, "/www");
"/ap" directory only include the file "index.htm" with header links to "/jquery/..". "/www" directory include all assets including "index.htm" with header links to the CDN.